### ISSUE TITLE
[BUG] Unclear exception when running com.azure.core.test.TestBase tests in PLAYBACK mode and the recorded-session file is missing

### ISSUE THREAD

**Issue by rabee333** [Azure/azure-sdk-for-java#6058]
**Describe the bug**
When running `com.azure.core.test.TestBase` tests in PLAYBACK mode, and the recorded-session file is missing, we get 2 confusing exception stack-traces: 1st stack-trace is on AssertionError, 2nd stack-trace is on NullPointerException.

***Exception or Stack Trace***
```
java.lang.AssertionError
	at org.junit.Assert.fail(Assert.java:87)
	at org.junit.Assert.fail(Assert.java:96)
	at com.azure.core.test.TestBase.setupTest(TestBase.java:52)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:564)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
...


java.lang.NullPointerException
	at com.azure.core.test.TestBase.teardownTest(TestBase.java:65)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:564)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
...
```

**To Reproduce**
- Extend com.azure.core.test.TestBase to produce a test with some http request.
- Run the test in PLAYBACK mode (Environment variable: AZURE_TEST_MODE=PLAYBACK)

**Expected behavior**
Print a clear message that the recorded session file was not found in the expected path.

**Setup (please complete the following information):**
 - OS: Windows, Linux, iOS
 - IDE : IntelliJ
 - Version of the Library used: jdk 1.8.0_221, jdk 9.0.4

**Additional context**

**Information Checklist**
Kindly make sure that you have added all the following information above and checkoff the required fields otherwise we will treat the issuer as an incomplete report
- [x] Bug Description Added
- [x] Repro Steps Added
- [x] Setup information Added

**Comment by joshfree** [comment]
@sima-zhu could you please assist?

**Comment by sima-zhu** [comment]
@rabee333 To make the playback run, you have to set env variable AZURE_TEST_MODE=RECORD. After generating the test json file under session-records, then you can run playback tests with AZURE_TEST_MODE=PLAYBACK. Let me know if you have more questions.

**Comment by rabee333** [comment]
@sima-zhu Yes, this is the way how to do a RECORD and then a PLAYBACK. 
The issue is when the json file is missing or with an incorrect name, when running a PLAYBACK, the test fails, and the output is unclear.
I think, in this case it should print an appropriate message describing that the file is not existing under the recorded-sessions.

**Comment by sima-zhu** [comment]
The fix has been checked in [`750ab29`](https://github.com/Azure/azure-sdk-for-java/commit/750ab299d23fe9b09daf17b46b4a60fab747af3c)

Thanks for reporting the issue!

Close the issue, please feel free to reopen it if it is still a problem.
